pennylane/qchem/tapering.py

```diff
@@ -484,14 +483,12 @@ def taper_hf(generators, paulixops, paulix_sector, num_electrons, num_wires):
     fermop_terms = []
     for idx, bit in enumerate(hf):
         if bit:
-            op_coeffs, op_terms = jordan_wigner([idx])
-            op_term = qml.Hamiltonian(np.array(op_coeffs), op_terms)
+            ps = qml.jordan_wigner(qml.FermiC(idx), ps=True)
         else:
-            op_term = qml.Hamiltonian([1.0], [qml.Identity(idx)])
-        fermop_terms.append(op_term)
+            ps = PauliSentence({PauliWord({idx: "I"}): 1.0})
+        fermop_terms.append(ps)
 
-    fermop_terms_as_ps = [pauli_sentence(term) for term in fermop_terms]
-    ferm_ps = functools.reduce(lambda i, j: i * j, fermop_terms_as_ps)
+    ferm_ps = functools.reduce(lambda i, j: i * j, fermop_terms)
```

Comment:
You could probably re-write this so that you are always adding PauliSentences. Something like: 

```
ferm_ps = PauliSentence({PauliWord({0: "I"})}: 0.0)
for idx, bit in enumerate(hf):
    # ... get ps ... 
    # ferm_ps *= ps 
```

Reply:
I think you meant `PauliSentence({PauliWord({0: "I"}): 1.0})`?